h2/src/test/org/h2/test/scripts/testScript.sql

```diff
@@ -2192,7 +2192,7 @@ drop table test;
 create domain email as varchar(200) check (position('@' in value) > 1);
 > ok
 
-create domain gmail as email default '@gmail.com' check (position('gmail' in value) > 1);
+create domain gmail as varchar(200) default '@gmail.com' check (position('gmail' in value) > 1);
```

Comment:
why can domains not reference other domains?

Reply:
The SQL Standard does not allow that.

Reply:
We can support them if you think that there is a good reason for such domains.

Reply:
I think you're taking this "SQL standard" thing a little too far - it is not gospel, it is just a baseline specification. We're quite free to support stuff that goes above and beyond that, and removing a useful feature because the standard does not mention it, does not seem useful.

Now if you're disallowing domains that reference __themselves__, that I'm fine with.

Reply:
H2 supported them in a strange way. They were dropped by a parser, when I fixed the parser for the `CAST` function their definitions were changed.

OK, I'll restore their support, but they will preserve the base domain from now.

Reply:
Now domains based on other domains preserve that information (due to changes for `CAST`).

But `DEFAULT` expressions are still copied from the parent domain and we also don't report the parent domains in a metadata, so we can't find their constraints here. That's not good.

This non-standard feature needs more design decisions.

Reply:
In the Standard user-defined data types may inherit each other, but H2 does not support them. In old versions of H2 domains were described as user-defined types, but they always were implemented as domains.